src/utilities.cpp

```diff
@@ -693,4 +695,70 @@ std::string makeUniqueId(IdList &idList)
     return id;
 }
 
+void linkComponentVariableUnits(const ComponentPtr &component, const LoggerPtr &logger)
+{
+    for (size_t index = 0; index < component->variableCount(); ++index) {
+        auto v = component->variable(index);
+        auto u = v->units();
+        if (u != nullptr) {
+            auto model = owningModel(u);
+            if (model == nullptr && !isStandardUnit(u)) {
```

Comment:
Should this raise an issue when `model != nullptr` ? At present it just skips over it?